### ISSUE TITLE
Instance.reload() breaking changes 2.x -> 3.x

### ISSUE THREAD

**Issue by defunctzombie** [sequelize/sequelize#3976]
It seems that it is no longer possible to reload only a specific field when calling `instance.reload()`. I use to accomplish this by passing an array of fields to reload to `reload()`. I attempted to follow the docs and pass `{ attributes: ['field_to_reload'] }` however when performing a subsequent save and specifying a single attribute to save I am presented with the following error:

> You attempted to save an instance with no primary key, this is not allowed since it would result in a global update

This leads me to believe that the instance reload for a specific attribute either 1) did not work or 2) reloaded that attribute but cleared out the previous values for other attributes.

If it is not possible to reload only a specific field that is fine (just worth noting in the reload docs). If it is desired functionality (rare but useful since in my case I know I just want a fresher version of a single field) then this seems like a bug.

Additionally, might be worth noting in the 3.x changelog that instance reload is another place there breaking changes might happen since the arguments from it are passed to find.

**Comment by mickhansen** [comment]
`fields: []` should still be supported. If it isn't we had a regression.
I imagine that we call `set` with `reset: true` with the new data values and since you only select a subset it whipes out the entire dataset and only sets the single field, should be simple enough to fix (only `reset` if `fields` is undefined`)

**Comment by defunctzombie** [comment]
Ok, `fields` does work. The reason I had not tried that (I was actually using `attributes`) was because the [reload documentation](http://sequelize.readthedocs.org/en/latest/api/instance/#reloadoptions-promisethis) said that the options were passed to `Model.find` so I went and looked at the [findAll documentation](http://sequelize.readthedocs.org/en/latest/api/model/#findalloptions-promisearrayinstance) and that only mentioned the `attributes` field. `fields` only appears in the `create` documentation it seems.

tl;dr; documentation bug/clarity issues around what options are available to the `reload` method.

**Comment by mickhansen** [comment]
Hmm you are right `fields` is actually only for `create'.
`attributes` should work for reload/findOne, but yeah i imagine it doesn't.

**Comment by defunctzombie** [comment]
`attributes` does not work. And now that I think about it, `fields` "working" might have just been that it was ignored and the whole model was loaded.

**Comment by mickhansen** [comment]
Yeah i think that might have been the case aswell. Definitely a regression.

**Comment by mickhansen** [comment]
Fixed in 3.3.1

**Comment by defunctzombie** [comment]
Thanks!
